catchup/service.go

```diff
@@ -464,3 +491,55 @@ func (s *Service) sync() {
 
 	s.log.Infof("Catchup Service: finished catching up, now at round %v (previously %v). Total time catching up %v.", s.ledger.LastRound(), pr, elapsedTime)
 }
+
+// nextRoundIsNotApproved returns true if the next round upgrades to a protocol version
+// which is not approved.
+// In case of an error, it returns false
+func (s *Service) nextRoundIsNotApproved(nextRound basics.Round) bool {
+	lastLedgerRound := s.ledger.LastRound()
+	proto, err := s.ledger.ConsensusParams(lastLedgerRound)
+	if err != nil {
+		s.log.Errorf("catchup: could not get consensus parameters for round %v: $%v", lastLedgerRound, err)
+		s.log.Errorf("catchup: could not determine if next round is from an approved protocol.")
+		return false
+	}
+	approvedUpgrades := proto.ApprovedUpgrades
+
+	block, error := s.ledger.Block(lastLedgerRound)
+	if error != nil {
+		s.log.Errorf("catchup: could not last block (%v) from the ledger.", lastLedgerRound)
```

Comment:
few notes:
1. replace the `catchup` with `nextRoundIsNotApproved`
2. fix the error message : "could not retrieve last block (%d) from ledger"

Reply:
dropping the second error log to reduce verbosity: 
could not determine if next round is from an approved protocol.